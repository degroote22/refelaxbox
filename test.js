require('browser-env')()
const test = require('ava')
const React = require('react')
const render = require('react-test-renderer').create
const {
  css,
  config,
  ReflexProvider,
  Box,
  Flex
} = require('./src')

const fela = require('fela')
const renderer = fela.createRenderer()
const ReactFela = require('react-fela')
const Provider = ReactFela.Provider

const props = {
  children: 'Hello',
  title: 'Boop',
  w: [1, 1 / 2, 1 / 4],
  m: 2,
  flex: true,
  px: 3
}

const cx = css({ ...config, renderer })

test.afterEach(t => {
  renderer.clear()
})

test('css returns props', t => {
  const a = cx(props)
  t.is(typeof a, 'object')
})

test('css passes props through', t => {
  const a = cx(props)
  t.is(a.children, 'Hello')
  t.is(a.title, 'Boop')
})

test('css strips style props', t => {
  const a = cx(props)
  t.is(a.w, undefined)
  t.is(a.m, undefined)
  t.is(a.flex, undefined)
  t.is(a.px, undefined)
})

test('css adds a className prop', t => {
  const a = cx(props)
  t.is(typeof a.className, 'string')
})

test('css inserts styles to sheet', t => {
  const a = cx({ m: 2 })
  const rules = renderer.renderToString()
  t.snapshot(rules)
})

test('css inserts responsive styles', t => {
  const a = css({ ...config, renderer })({ m: [1, 2] })
  const rules = renderer.renderToString()
  t.snapshot(rules)
})

test('css dedupes repeated styles', t => {
  const a = cx({ p: 2 })
  const b = cx({ p: 2 })
  const rules = renderer.renderToString()
  t.snapshot(rules)
  t.deepEqual(a, b)
})

test('css parses widths', t => {
  const a = cx({ w: 1 })
  const b = cx({ w: 1 / 2 })
  const c = cx({ w: 0 })
  const d = cx({ w: 24 })
  const e = cx({ w: 'auto' })
  const rules = renderer.renderToString()
  t.snapshot(rules)
})

test('css parses margins', t => {
  cx({ m: 1 })
  cx({ m: -2 })
  cx({ m: 24 })
  cx({ m: -24 })
  cx({ m: 'auto' })
  const rules = renderer.renderToString()
  t.snapshot(rules)
})

test('css parses margin directions', t => {
  cx({ m: 1 })
  cx({ mt: 1 })
  cx({ mr: 1 })
  cx({ mb: 1 })
  cx({ ml: 1 })
  cx({ mx: 1 })
  cx({ my: 1 })
  const rules = renderer.renderToString()
  t.snapshot(rules)
})

test('css parses paddings', t => {
  cx({ p: 1 })
  cx({ p: 24 })
  cx({ p: '20%' })
  const rules = renderer.renderToString()
  t.snapshot(rules)
})

test('css parses padding directions', t => {
  cx({ p: 1 })
  cx({ pt: 1 })
  cx({ pr: 1 })
  cx({ pb: 1 })
  cx({ pl: 1 })
  cx({ px: 1 })
  cx({ py: 1 })
  const rules = renderer.renderToString()
  t.snapshot(rules)
})

test('css parses flexbox styles', t => {
  cx({
    flex: true,
    wrap: true,
    column: true,
    auto: true,
    order: 5,
    align: 'center',
    justify: 'space-between'
  })
  const rules = renderer.renderToString()
  t.snapshot(rules)
})

test('snapshot and fela provider', t => {
  const box = render(<Provider renderer={renderer}><Box /></Provider>).toJSON()
  const flex = render(<Provider renderer={renderer}><Flex /></Provider>).toJSON()
  t.snapshot(box)
  t.snapshot(flex)
})

test('react context for reflex and fela provider', t => {
  const ctx = {
    space: [0, 6, 12, 18, 24, 30],
    breakpoints: [32, 48, 64]
  }
  const box = render(
    <Provider renderer={renderer}>
      <ReflexProvider {...ctx}>
        <Box m={1} w={[1, 1 / 2]} />
      </ReflexProvider>
    </Provider>
  ).toJSON()
  t.snapshot(box)
  const rules = renderer.renderToString()
  t.snapshot(rules)
})
