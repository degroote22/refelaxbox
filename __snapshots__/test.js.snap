// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`css dedupes repeated styles 1`] = `".a{padding:16px}"`;

exports[`css inserts responsive styles 1`] = `".a{margin:8px}@media screen and (min-width:40em){.b{margin:16px}}"`;

exports[`css inserts styles to sheet 1`] = `".a{margin:16px}"`;

exports[`css parses flexbox styles 1`] = `".a{display:flex}.b{flex-wrap:wrap}.c{flex-direction:column}.d{flex:1 1 auto}.e{order:5}.f{align-items:center}.g{justify-content:space-between}"`;

exports[`css parses margin directions 1`] = `".a{margin:8px}.b{margin-top:8px}.c{margin-right:8px}.d{margin-bottom:8px}.e{margin-left:8px}"`;

exports[`css parses margins 1`] = `".a{margin:8px}.b{margin:-16px}.c{margin:24px}.d{margin:-24px}.e{margin:auto}"`;

exports[`css parses padding directions 1`] = `".a{padding:8px}.b{padding-top:8px}.c{padding-right:8px}.d{padding-bottom:8px}.e{padding-left:8px}"`;

exports[`css parses paddings 1`] = `".a{padding:8px}.b{padding:24px}.c{padding:20%}"`;

exports[`css parses widths 1`] = `".a{width:100%}.b{width:50%}.c{width:0%}.d{width:24px}.e{width:auto}"`;

exports[`react context 1`] = `
<div
  className="a b c"
/>
`;

exports[`react context 2`] = `".a{margin:6px}.b{width:100%}@media screen and (min-width:32em){.c{width:50%}}"`;

exports[`react context for reflex and fela provider 1`] = `
<div
  className="a b c"
/>
`;

exports[`react context for reflex and fela provider 2`] = `".a{margin:6px}.b{width:100%}@media screen and (min-width:32em){.c{width:50%}}"`;

exports[`snapshot 1`] = `
<div
  className=""
/>
`;

exports[`snapshot 2`] = `
<div
  className="a"
/>
`;

exports[`snapshot and fela provider 1`] = `
<div
  className=""
/>
`;

exports[`snapshot and fela provider 2`] = `
<div
  className="a"
/>
`;
